File: python/sorting/selection_sort.py
from .sort import SortType, Sort


class SelectionSort(Sort):
    def sort(self, arr: list[int], type=SortType.asc):
        length: int = len(arr)

        for i in range(length - 1):
            min_index = i
            for j in range(i + 1, length):
                if super().compare(arr[min_index], arr[j], type):
                    min_index = j

            arr[i], arr[min_index] = arr[min_index], arr[i]

        return arr

File: typescript/sorting/selection_sort.ts
import Sort, { SortType, SORT_TYPE } from "./sort.ts";

class SelectionSort extends Sort {
  sort(arr: number[], type: SortType = SORT_TYPE.asc): number[] {
    const length: number = arr.length;

    for (let i = 0; i < length - 1; i++) {
      let minIndex: number = i;
      for (let j = i + 1; j < length; j++) {
        if (super.compare(arr[minIndex], arr[j], type)) {
          minIndex = j;
        }
      }
      [arr[minIndex], arr[i]] = [arr[i], arr[minIndex]];
    }
    return arr;
  }
}

export default SelectionSort;
